src/addons/transitions/ReactCSSTransitionGroupChild.js

```diff
@@ -48,8 +48,8 @@ var ReactCSSTransitionGroupChild = React.createClass({
 
   transition: function(animationType, finishCallback) {
     var node = React.findDOMNode(this);
-    var className = this.props.name + '-' + animationType;
-    var activeClassName = className + '-active';
+    var className = this.props.name[animationType] || this.props.name + '-' + animationType;
+    var activeClassName = this.props.name[animationType + 'Active'] || className + '-active';
```

Comment:
I can't seem to get tests to work :/ any suggestions on how I should approach it? There aren't currently any tests to make sure the classnames are correct.